File: 04 Neo4J/backend/models/userModel.js
class User {
    constructor(username, password) {
      this.username = username;
      this.password = password;
    }
  
    static async create(username, password) {
      const session = require('../config/db').session;
      const result = await session.run(
        'CREATE (u:User {username: $username, password: $password}) RETURN u',
        { username, password }
      );
      return result.records[0].get('u');
    }
  
    static async findByUsername(username) {
      const session = require('../config/db').session;
      const result = await session.run(
        'MATCH (u:User {username: $username}) RETURN u',
        { username }
      );
      return result.records.length > 0 ? result.records[0].get('u') : null;
    }
  
    static async updatePassword(username, newPassword) {
      const session = require('../config/db').session;
      const result = await session.run(
        'MATCH (u:User {username: $username}) SET u.password = $newPassword RETURN u',
        { username, newPassword }
      );
      return result.records[0].get('u');
    }
  
    static async delete(username) {
      const session = require('../config/db').session;
      await session.run(
        'MATCH (u:User {username: $username}) DELETE u',
        { username }
      );
    }
  }
  
  module.exports = User;
  

File: 04 Neo4J/Python/backend/models/user_model.py
from config.db import conn

class User:
    @staticmethod
    def create(username, password):
        query = '''
        CREATE (u:User {username: $username, password: $password})
        RETURN u
        '''
        conn.query(query, parameters={'username': username, 'password': password})

    @staticmethod
    def find_by_username(username):
        query = '''
        MATCH (u:User {username: $username})
        RETURN u
        '''
        result = conn.query(query, parameters={'username': username})
        return result[0]['u'] if result else None

    @staticmethod
    def update_password(username, new_password):
        query = '''
        MATCH (u:User {username: $username})
        SET u.password = $new_password
        RETURN u
        '''
        conn.query(query, parameters={'username': username, 'new_password': new_password})

    @staticmethod
    def delete(username):
        query = '''
        MATCH (u:User {username: $username})
        DELETE u
        '''
        conn.query(query, parameters={'username': username})
